### ISSUE TITLE
[Bug] - REST ListObjectsV2 - continuation tokens behave like v1 markers

### ISSUE THREAD

**Issue by lrm25** [versity/versitygw#995]
**Describe the bug**
From v1 to v2, AWS shifted away from markers to continuation tokens.  These have different behavior, in that they're generated, rather than just going through the list in alphabetical order.  versitygw just mimics v1 marker behavior with the continuation token elements rather than following the v2 protocol.

**To Reproduce**
See #993.

**Expected behavior**
v2 continuation tokens behave the same as direct to s3 continuation tokens.

**Server Version**
Version  : v1.0.9
Build    : c0b6183
BuildTime: 2024-12-20_07:07:55PM
Darwin Lukes-MacBook-Pro.local 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:25 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6030 arm64

**Comment by niksis02** [comment]
The implementation of ListObjectsV2's ContinuationToken in the gateway differs from the one in S3. It is more similar to the NextMarker used in ListObjects.

In S3, the ContinuationToken is generated based on object keys, whereas the gateway directly returns the object key as the ContinuationToken.

I'm unsure if this needs to be changed—@benmcclelland might be able to clarify.

**Comment by lrm25** [comment]
Yeah, I'm not sure either.  I looked up that the two main reasons that AWS made this switch was to prevent clients from skipping parts of the list by adding the wrong key to the marker, and to prevent people from trying to query files they didn't have access to.  I added the bug just in case.
